.circleci/config.yml

```diff
@@ -49,14 +51,14 @@ jobs:
               rm -f dot.bin
             fi
 
-      - run:
-          # We use the python Pipfile.lock as the cache key and also
-          # the hash of the current python.
-          # If dependencies haven't changed then use the cache.
-          name: Create Python cache key
+      - run: &create_python_cache_key
+          # Combine hashes of the Python interpreter and Pipfile plus today's
+          # date into a file whose checksum will key the Python environ. cache.
+          name: Create Python environment cache key
           command: |
-            md5sum lib/Pipfile.locks/${CIRCLE_JOB} > ~/python_version.md5
-            md5sum $(which python) >> ~/python_version.md5
+            md5sum $(which python) > ~/python_version.md5
+            md5sum lib/Pipfile >> ~/python_version.md5
```

Comment:
As above, the CI job will now once per day solve for the latest versions of our dependencies that are compatible with each other and our specification in `lib/Pipfile`. The lockfiles were *only* used by CircleCI, so our PR-based testing flow could unexpectedly break, but on the other hand we'd be less likely to have a break in the wild that we weren't aware of because we were happily using pinned versions.